### ISSUE TITLE
apop and lpop checks with Oracle tables

### ISSUE THREAD

**Issue by zoyafuso-NOAA** [BenWilliams-NOAA/swo#40]
Hi, I hope this is a good place to put this question. Could you summarize the level of testing you have done with the output from the apop and lpop functions and the corresponding output tables in Oracle, e.g., GOA.SIZECOMP_STRATUM, GOA.AGECOMP_TOTAL, HAEHNR.SIZECOMP_STANDARD_STRATUM and HAEHNR.AGECOMP_EBS_STANDARD_STRATUM? I have been going through the process of consolidating the size/age comp code used by the BS and AIGOA groups and I am finding that are some differences between how each group calculates our indices/comps. Going through some of the functions in this repo, I'm wondering whether you've also seen these differences when testing your output among regions.

**Comment by pete-hulson** [comment]
Hey there - with swo I've focused (to this point) on matching apop and lpop for the GOA and AI surveys (we've done a prelim look at BS, but I'm planning on digging a bit further into that). Bottom line up front, we can match the GOA, but can't seem to match the AI pre-2006 (not sure what's going on there...). If you check out this issue [https://github.com/afsc-gap-products/data-requests/issues/8](url) I've tried to explain what's going on. But yes, we've also found differences in how the BS and AIGOA do things, and to deal with this we've got different functions for each region to do the apop/lpop calcs. Lemme know if this helps - I'll be working today but then on leave again till mid next week

**Comment by zoyafuso-NOAA** [comment]
Thanks Pete, that's good information to know, and I'll report back to my index computation group that these independent ways of calculating comps are revealing differences in how age/size comps are calculated among regions. Our index computation working group is hoping to make the index and comp calculations consistent among regions. 

Sorry our team has been slow addressing your issue (from back in Feb!). It seems we have similar issues so I'll comment on that post.

**Comment by BenWilliams-NOAA** [comment]
@zoyafuso-NOAA 

Thanks for reaching out with this. In addition to what Pete mentioned we also noticed differences between BSAI and GOA for how hauls where catch numbers exist but no lengths are observed are treated.
https://github.com/BenWilliams-NOAA/swo-tech/issues/11
